UnitsNet/UnitInfo.cs

```diff
@@ -72,6 +73,15 @@ public UnitInfo(Enum value, string pluralName, BaseUnits baseUnits, string quant
         /// Name of the quantity this unit belongs to. May be null for custom units.
         /// </summary>
         public string? QuantityName { get; }
+
+        /// <summary>
+        ///     Gets the unique key representing the unit type and its corresponding value.
+        /// </summary>
+        /// <remarks>
+        ///     This key is particularly useful when using an enum-based unit in a hash-based collection,
+        ///     as it avoids the boxing that would normally occur when casting the enum to <see cref="Enum" />.
+        /// </remarks>
+        public virtual UnitKey UnitKey => Value;
```

Comment:
I like the new `UnitKey`.

Reply:
> Agreed, but why not just remove it since this is targeting v6?
We could deprecate in v5 instead and offer a similar change there.

I was hoping you would say that (regarding the `QuantityInfo.ValueType`)..